ci/pipeline.yaml

```diff
@@ -233,3 +248,64 @@ jobs:
                 ! diff --context=3 \
                   <(tar xzfO nginx-chart/*.tgz nginx/values.yaml) \
                   <(tar xzfO rewritten-chart/chart.tgz nginx/values.yaml)
+
+
+  - name: test-pull-request
+    serial: true
+    plan:
+      - in_parallel:
+        - get: test-image
+          passed:
+            - build-test-image
+        - get: source
+          resource: pull-requests
+          trigger: true
+          version: every
+          params:
+            integration_tool: rebase
+      - put: pull-requests
+        inputs: detect
+        params:
+          path: source
+          status: pending
+      - task: run-unit-tests
+        image: test-image
+        file: source/ci/tasks/test.yaml
+        params:
+          MAKE_TARGET: test-units
+        on_failure:
+          put: pull-requests
+          inputs: detect
+          params:
+            path: source
+            status: failure
+      - task: run-feature-tests
+        image: test-image
+        file: source/ci/tasks/test.yaml
+        params:
+          MAKE_TARGET: test-features
+        on_failure:
```

Comment:
is there any way in concourse to not to repeat this structure over and over? 

```
 on_failure:
          put: pull-requests
          inputs: detect
          params:
            path: source
            status: failure
```

Reply:
Yeah, I think so. Let me try something.